diff --git a/dist/codegen/common.js b/dist/codegen/common.js
index 411d88673e1edf8c12ecb0891c0d65246070c61e..204935d443b70d39adac01971967d9d7f244dd78 100644
--- a/dist/codegen/common.js
+++ b/dist/codegen/common.js
@@ -135,7 +135,8 @@ const lexiconsTs = (project, lexicons) => gen(project, '/lexicons.ts', async (fi
         return nsid
             .split('.')
             .map((word) => word[0].toUpperCase() + word.slice(1))
-            .join('');
+            .join('')
+			.replaceAll("-", "_");
     };
     //= import { type LexiconDoc, Lexicons } from '@atproto/lexicon'
     file
